### ISSUE TITLE
Install mongoclient as a global npm package

### ISSUE THREAD

**Issue by michalvankodev** [nosqlclient/nosqlclient#58]
It would be super cool if I would be able to install mongoclient just as
`npm install -g mongoclient`.

So I could have just install upgrades seamlessly with
`npm update -g` and always have the newest of the best.

I tried to enable it with setting meteor just as dependency in `package.json` but I've failed to install it because meteor doesn't support newer versions of node (>0.10.x).

If you like this concept. We could left this story here and when meteor could be easily installed as a dependency of your package I could implement this.

**Comment by rsercano** [comment]
That would be awesome. A very nice and interesting idea, maybe you should have a look at  meteor 1.3. Since I don't have enough time migrating to 1.3 is still in the queue, if it helps you I can accelerate this issue. #37

**Comment by michalvankodev** [comment]
Well.. the Meteor not being able to install on newer node is also a deal breaker so there are plenty of thing that have to happen so this could be possible to do

**Comment by rsercano** [comment]
I'll just leave this here for further reference :)

**Comment by dynnamitt** [comment]
+1

**Comment by Droppix** [comment]
+1

**Comment by ghost** [comment]
@michalvankodev, I think that is a very cool idea, and now that meteor actually support newer node versions such as 4.4.7 would it be beneficial to update this repository.

I have sent a pull request updating mongoclient to meteor 1.3.5.1 (without the ecmascript package though).

The next step is to:
- activate the ecmascript atmosphere package
- activate the meteor-node-stubs npm package
- load npm modules though "import" instead of "Meteor.npmRequire"

@michalvankodev, you seem to be quite npm savvy, and I got into trouble, so would you be interested in taking a look at that?

I'm afraid these steps have to be taken before the repository can be updated to meteor 1.4 and hence node 4.4.7.

**Comment by rsercano** [comment]
@birkskyum I would like to accept pull requests for these mentioned features :)

**Comment by michalvankodev** [comment]
@birkskyum Awesome to hear about interest and progress on this issue. :) I would be glad to help in any way. :)

**Comment by ghost** [comment]
Great! 

The pull request I made 'Update meteor version' contains a version of mongoclient that's build on Meteor 1.4. It is finally running, but there are some visual regression because it rely on some deprecated atmosphere packages.

The atmosphere packages introduced by this pull request are:

**ephemer:reactive-datatables**
chrismbeckett:toastr
**joshdellay:meteor-ladda-bootstrap**
tkornblit:icheck
perak:codemirror
flot:flot
mrt:chosen-package
linto:jquery-ui
kit:sweetalert
rochal:slimscroll
xvendo:clipboardjs
onokumus:metismenu
sujith3g:jstree

It would be very nice to load these directly from npm instead, and especially the two packages marked as bold are challenging because it is old versions. @michalvankodev Would you be up for trying to load some of these from npm instead?

**Comment by michalvankodev** [comment]
With npm it shouldn't be a problem to depend on older versions of packages, because you can specify exact version of package your package depends on.
I've never worked with meteor before so I don't exactly know how atmosphere works.

**Comment by ghost** [comment]
@michalvankodev It's actually the other way around. The project is right now forced to use deprecated versions of several 3rd-party libraries because the atmosphere packages aren't updated. If the libraries are loaded through npm would it be possible to use the newest versions of the libraries, which would solve the visual regression issues.

In meteor are all files in /client automagically loaded into the client, and it can make it quite hard to control the load order. Many atmosphere packages (such as this one: https://github.com/JoshDellay/meteor-ladda-bootstrap/) simply dump the library files into '/client', which can be seen in package.js. It would be better to load it through npm instead, so it isn't a problem that you aren't familiar with atmosphere.

**Comment by michalvankodev** [comment]
Ah.. I'm looking at the meteor right now and it looks like this issue will not be able to be solved at all as meteor is wrapping it's own node runtime. What I wanted was use mongoclient as a package for node itself but meteor is really standing in a way.

**Comment by michalvankodev** [comment]
@birkskyum I've forked your branch but I still fail to install meteor as a classic package on my node version. With your atmoshere packages I can suggest that you remove them from atmosphere package.json like file and install them with npm instead.. but I don't think it will resolve all of the visual issues found in your PR.

**Comment by birkskyum** [comment]
Maybe not, but if the packages are loaded through npm can I fix the visuals easier.

**Comment by ghost** [comment]
@michalvankodev A pr upgrading Mongoclient to 1.4 (node 4.4.7) was merged today. Does that solve the issue you initially met related to node 0.10.x?

**Comment by michalvankodev** [comment]
@birkskyum The issue was not with node 0.10.x but with the meteor itself. Meteor is a platform but not a package. You can't install meteor as a dependency for another package therefore I can't make mongoclient package as a global package which will depend on meteor.

**Comment by rsercano** [comment]
@michalvankodev so this is impossible for near future ? Or would something like https://github.com/onmodulus/demeteorizer works ?

**Comment by ghost** [comment]
Yes, the 1.5 release of Meteor was previously assigned to move Meteor towards npm, but it was changed to include Apollo/Meteor integration instead.

It is on meteor's roadmap to move towards npm, but they haven't announced anything about how it is going to work.

**Comment by rsercano** [comment]
I see, thanks for clarifying @birkskyum

**Comment by bsbechtel** [comment]
+1 for this issue. I just found out about this project. I'd like to add it as a package to run inside my current meteor application instead of running it as a separate application. To do that, it would need to be integrated as an npm or meteor/atmosphere package.

**Comment by rsercano** [comment]
@bsbechtel it seems to be hard to arrange it as npm package, but at least will try to do it as an atmosphere package soon

**Comment by bsbechtel** [comment]
That would be great! I think all it would require is repackaging the code and exporting a few variables using package.js.

**Comment by rsercano** [comment]
@birkskyum Hi again, since meteor 1.6 uses node 8+ and mongoclient master uses meteor 1.6, is there a possibility to make this happen ?
